.github/workflows/publish.yml

```diff
@@ -2,60 +2,56 @@ name: Publish
 
 on:
   push:
-    branches: ["main"]
-    tags: ["*.*.*"]
-    paths-ignore: ["*.md"]
+    branches:
+    - main
 
-env:
-  CI: true
-
-# Pnpm setup based on https://github.com/pnpm/action-setup#use-cache-to-reduce-installation-time
+concurrency: ${{ github.workflow }}-${{ github.ref }}
 
 jobs:
-  version:
+  release:
+    name: Release
     runs-on: ubuntu-latest
     permissions:
       id-token: write # required for provenance https://docs.npmjs.com/generating-provenance-statements#publishing-packages-with-provenance-via-github-actions
       contents: write
       pull-requests: write
 
     steps:
-      - name: Checkout
-        uses: actions/checkout@v4
-        with:
-          fetch-depth: 0
+      - uses: actions/checkout@v4
 
       - name: Install Node.js
         uses: actions/setup-node@v4
+        with:
+          node-version-file: ".nvmrc"
+          cache: "pnpm"
 
       - name: Install pnpm
         uses: pnpm/action-setup@v4
-        id: pnpm-install
         with:
           version: 9
           run_install: false
 
       - name: Get pnpm store directory
-        id: pnpm-cache
         shell: bash
         run: |
-          echo "STORE_PATH=$(pnpm store path)" >> $GITHUB_OUTPUT
+          echo "STORE_PATH=$(pnpm store path --silent)" >> $GITHUB_ENV
 
       - name: Setup pnpm cache
         uses: actions/cache@v4
         with:
-          path: ${{ steps.pnpm-cache.outputs.STORE_PATH }}
+          path: ${{ env.STORE_PATH }}
           key: ${{ runner.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
           restore-keys: |
             ${{ runner.os }}-pnpm-store-
 
       - name: Install Dependencies
         run: pnpm i --frozen-lockfile
 
-      - name: Set publishing config
-        run: pnpm config set '//registry.npmjs.org/:_authToken' "${NODE_AUTH_TOKEN}"
+      - name: Create Release Pull Request or Publish to npm
+        id: changesets
+        uses: changesets/action@v1
+        with:
+          publish: pnpm run changeset:publish
```

Comment:
also for my own understanding, is there a concept of preview packages for npm or will publish official versions only?

Reply:
Right now, only official versions will be published.

[Prereleases (ex. `1.0.0-beta`) are supported by changeset](https://github.com/changesets/changesets/blob/main/docs/prereleases.md) but they are very complex.

Changeset also supports [snapshot releases](https://github.com/changesets/changesets/blob/main/docs/snapshot-releases.md), which will release your package with version `0.0.0-{name}-{timestamp}` which you can use for testing before publishing the package for real. It's something I've been meaning to explore for a while, and I think it would be worth creating a ticket in the "nice to have" for it.